jupyter_notebook_config.py

```diff
@@ -76,3 +78,49 @@
             sys.exit(155)
     else:
         print('Tags variable is not an array.', file=sys.stderr)
+
+def saveFile(file_path, token):
+    """
+    Construct a requests POST call with args and kwargs and process the
+    results.
+    Args:
+        file_path: The relative path to the file from the datadir, including filename and extension
+        token: keboola storage api token
+    Returns:
+        body: Response body parsed from json.
+    Raises:
+        requests.HTTPError: If the API request fails.
+    """
+
+    url = 'http://data-loader-api/data-loader-api/save'
+    headers = {'X-StorageApi-Token': token, 'User-Agent': 'Keboola Sandbox Autosave Request'}
+    payload = {}
+    payload['file'] = {'source': file_path, 'tags': ['autosave']}
+
+    r = requests.post(url, json=payload, headers=headers, timeout=240)
+    try:
+        r.raise_for_status()
+    except requests.HTTPError:
+        # Handle different error codes
+        raise
+    else:
+        return r.json()
+
+def script_post_save(model, os_path, contents_manager, **kwargs):
+    """
+    saves the ipynb file to keboola storage on every save within the notebook
+    """
+    if model['type'] != 'notebook':
+        return
+    log = contents_manager.log
+
+    # get the token from env
+    token = None
+    if 'KBC_TOKEN' in os.environ:
+        token = os.environ['KBC_TOKEN']
+    else:
+        log.error("Could not find the keboola api token.")
+    response = saveFile(os.path.relpath(os_path), token)
```

Comment:
response is not used

Reply:
also what happens when saveFile throws exception? shouldn't we catch it and log it?

Reply:
when it throws any exception it is displayed in the console in an orange bar here:
![image](https://user-images.githubusercontent.com/569139/73547117-9cf22f00-443e-11ea-92dd-e8f76d36d55c.png)
should probably throw an exception as well as log the error though, also for the no token found case.

Reply:
definitely - make that consistent